br full backup need to report correct error code when minio is full

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
1、run workload with multi mvcc
2、enable IME
3、br full backup

### 2. What did you expect to see? (Required)
br full backup can succeed

### 3. What did you see instead (Required)
br full backup failed
`[2025/01/02 11:32:09.287 +00:00] [ERROR] [backup.go:58] ["failed to backup"] [error="error happen in store 5: unknown error, retried too many times, give up: [BR:KV:ErrKVStorage]tikv storage occur I/O error"] [errorVerbose="[BR:KV:ErrKVStorage]tikv storage occur I/O error\nerror happen in store 5: unknown error, retried too many times, give up\ngithub.com/pingcap/tidb/br/pkg/backup.(*Client).OnBackupResponse\n\t/workspace/source/tidb/br/pkg/backup/client.go:1213\ngithub.com/pingcap/tidb/br/pkg/backup.(*Client).RunLoop\n\t/workspace/source/tidb/br/pkg/backup/client.go:341\ngithub.com/pingcap/tidb/br/pkg/backup.(*Client).BackupRanges\n\t/workspace/source/tidb/br/pkg/backup/client.go:1126\ngithub.com/pingcap/tidb/br/pkg/task.RunBackup\n\t/workspace/source/tidb/br/pkg/task/backup.go:689\nmain.runBackupCommand\n\t/workspace/source/tidb/br/cmd/br/backup.go:57\nmain.newFullBackupCommand.func1\n\t/workspace/source/tidb/br/cmd/br/backup.go:149\ngithub.com/spf13/cobra.(*Command).execute\n\t/root/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:985\ngithub.com/spf13/cobra.(*Command).ExecuteC\n\t/root/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1117\ngithub.com/spf13/cobra.(*Command).Execute\n\t/root/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1041\nmain.main\n\t/workspace/source/tidb/br/cmd/br/main.go:36\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:272\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1700"] [stack="main.runBackupCommand\n\t/workspace/source/tidb/br/cmd/br/backup.go:58\nmain.newFullBackupCommand.func1\n\t/workspace/source/tidb/br/cmd/br/backup.go:149\ngithub.com/spf13/cobra.(*Command).execute\n\t/root/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:985\ngithub.com/spf13/cobra.(*Command).ExecuteC\n\t/root/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1117\ngithub.com/spf13/cobra.(*Command).Execute\n\t/root/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1041\nmain.main\n\t/workspace/source/tidb/br/cmd/br/main.go:36\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:272"]
[2025/01/02 11:32:09.287 +00:00] [ERROR] [main.go:38] ["br failed"] [error="error happen in store 5: unknown error, retried too many times, give up: [BR:KV:ErrKVStorage]tikv storage occur I/O error"] [errorVerbose="[BR:KV:ErrKVStorage]tikv storage occur I/O error\nerror happen in store 5: unknown error, retried too many times, give up\ngithub.com/pingcap/tidb/br/pkg/backup.(*Client).OnBackupResponse\n\t/workspace/source/tidb/br/pkg/backup/client.go:1213\ngithub.com/pingcap/tidb/br/pkg/backup.(*Client).RunLoop\n\t/workspace/source/tidb/br/pkg/backup/client.go:341\ngithub.com/pingcap/tidb/br/pkg/backup.(*Client).BackupRanges\n\t/workspace/source/tidb/br/pkg/backup/client.go:1126\ngithub.com/pingcap/tidb/br/pkg/task.RunBackup\n\t/workspace/source/tidb/br/pkg/task/backup.go:689\nmain.runBackupCommand\n\t/workspace/source/tidb/br/cmd/br/backup.go:57\nmain.newFullBackupCommand.func1\n\t/workspace/source/tidb/br/cmd/br/backup.go:149\ngithub.com/spf13/cobra.(*Command).execute\n\t/root/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:985\ngithub.com/spf13/cobra.(*Command).ExecuteC\n\t/root/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1117\ngithub.com/spf13/cobra.(*Command).Execute\n\t/root/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1041\nmain.main\n\t/workspace/source/tidb/br/cmd/br/main.go:36\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:272\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1700"] [stack="main.main\n\t/workspace/source/tidb/br/cmd/br/main.go:38\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:272"]`

### 4. What is your TiDB version? (Required)
sh-5.1# ./br -V
Release Version: v8.5.0
Git Commit Hash: d13e52ed6e22cc5789bed7c64c861578cd2ed55b
Git Branch: HEAD
Go Version: go1.23.3
UTC Build Time: 2024-12-18 02:28:02
Race Enabled: false

--- a/br/pkg/backup/client.go
+++ b/br/pkg/backup/client.go
@@ -1205,15 +1205,12 @@ func (bc *Client) OnBackupResponse(
 		res := utils.HandleBackupError(errPb, storeID, errContext)
 		switch res.Strategy {
 		case utils.StrategyGiveUp:
-			errMsg := res.Reason
-			if len(errMsg) <= 0 {
-				errMsg = errPb.Msg
-			}
 			logutil.CL(ctx).Error("TiKV encountered an error, interrupting the backup.", zap.String("error message", errPb.Msg))
 			// TODO output a precise store address. @3pointer
-			return nil, errors.Annotatef(berrors.ErrKVStorage, "error happen in store %v: %s",
+			return nil, errors.Annotatef(berrors.ErrKVStorage, "error happen in store %v: %s, %s",
 				storeID,
-				errMsg,
+				res.Reason,
+				errPb.Msg,
 			)
 		}
 	}

--- a/br/pkg/backup/client.go
+++ b/br/pkg/backup/client.go
@@ -1209,7 +1209,7 @@ func (bc *Client) OnBackupResponse(
 			if len(errMsg) <= 0 {
 				errMsg = errPb.Msg
 			}
-			logutil.CL(ctx).Error("the backup error of TiKV leads to an interruption of the backup process", zap.String("error message", errPb.Msg))
+			logutil.CL(ctx).Error("TiKV encountered an error, interrupting the backup.", zap.String("error message", errPb.Msg))
 			// TODO output a precise store address. @3pointer
 			return nil, errors.Annotatef(berrors.ErrKVStorage, "error happen in store %v: %s",
 				storeID,

--- a/br/pkg/backup/client.go
+++ b/br/pkg/backup/client.go
@@ -1209,7 +1209,7 @@ func (bc *Client) OnBackupResponse(
 			if len(errMsg) <= 0 {
 				errMsg = errPb.Msg
 			}
-			log.Warn("the backup error from tikv make br progress been given up", zap.String("error message", errPb.Msg))
+			logutil.CL(ctx).Error("the backup error from tikv make br progress been given up", zap.String("error message", errPb.Msg))
 			// TODO output a precise store address. @3pointer
 			return nil, errors.Annotatef(berrors.ErrKVStorage, "error happen in store %v: %s",
 				storeID,

--- a/br/pkg/backup/client.go
+++ b/br/pkg/backup/client.go
@@ -1209,7 +1209,7 @@ func (bc *Client) OnBackupResponse(
 			if len(errMsg) <= 0 {
 				errMsg = errPb.Msg
 			}
-			logutil.CL(ctx).Error("the backup error from tikv make br progress been given up", zap.String("error message", errPb.Msg))
+			logutil.CL(ctx).Error("the backup error of TiKV leads to an interruption of the backup process", zap.String("error message", errPb.Msg))
 			// TODO output a precise store address. @3pointer
 			return nil, errors.Annotatef(berrors.ErrKVStorage, "error happen in store %v: %s",
 				storeID,

--- a/br/pkg/backup/client.go
+++ b/br/pkg/backup/client.go
@@ -1209,6 +1209,7 @@ func (bc *Client) OnBackupResponse(
 			if len(errMsg) <= 0 {
 				errMsg = errPb.Msg
 			}
+			log.Warn("the backup error from tikv make br progress been given up", zap.String("error message", errPb.Msg))
 			// TODO output a precise store address. @3pointer
 			return nil, errors.Annotatef(berrors.ErrKVStorage, "error happen in store %v: %s",
 				storeID,

